storage/command_queue.go

```diff
@@ -74,6 +75,34 @@ func (cq *CommandQueue) onEvicted(key, value interface{}) {
 	}
 }
 
+// GetWaitBatch is like GetWait, but it makes sure that all the spans are added
+// to the command queue simultaneously. That is, one can add overlapping
+// commands without them blocking on each other (in which case the command
+// would be deadlocked).
+// TODO(tschottdorf): update comment and replace GetWait with this.
+func (cq *CommandQueue) GetWaitBatch(spans []keys.Span, readOnly bool, wg *sync.WaitGroup) {
+	var pending []*[]*sync.WaitGroup
```

Comment:
no, I need to change `c.pending` for various `c`s, so I think I need to actually do it like this. I could collect the `&c`s if you prefer that.